sypht/client.py

```diff
@@ -42,17 +44,13 @@ def __init__(self, client_id=None, client_secret=None, base_endpoint=None, auth_
 
         if client_id is None or client_secret is None:
             raise ValueError('Client credentials missing')
+            
         self.client_id = client_id
+        self._client_secret = client_secret
         self._company_id = None
-
-        auth_endpoint = auth_endpoint or os.environ.get('SYPHT_AUTH_ENDPOINT', SYPHT_AUTH_ENDPOINT)
-        if '/oauth/' in auth_endpoint:
-            self._access_token = self._authenticate_v1(auth_endpoint, client_id, client_secret, audience=self.audience)
-        elif '/oauth2/' in auth_endpoint:
-            self._access_token = self._authenticate_v2(auth_endpoint, client_id, client_secret, audience=self.audience)
-        else:
-            raise ValueError(f"Invalid authentication endpoint: {auth_endpoint}")
-
+        self._access_token, expires_in = self._authenticate_client()
```

Comment:
Suggest consolidating this state management into `self._authenticate()`, then you avoid the similar operation repeated down in `_get_headers` too.